gridfs/asynchronous/grid_file.py

```diff
@@ -1616,18 +1613,37 @@ async def read(self, size: int = -1) -> bytes:
         """
         return await self._read_size_or_line(size=size)
 
-    async def readline(self, size: int = -1) -> bytes:  # type: ignore[override]
+    async def readline(self, size: int = -1) -> bytes:
         """Read one line or up to `size` bytes from the file.
 
         :param size: the maximum number of bytes to read
         """
         return await self._read_size_or_line(size=size, line=True)
 
+    async def readlines(self, size: int = -1) -> list[bytes]:
```

Comment:
This should be async only too since we intentionally get it for free  from IOBase in the sync version.